Orange/widgets/data/oweditdomain.py

```diff
@@ -662,6 +673,168 @@ def on_label_selection_changed(self):
+        max_val = len(data)
+        spin2.setMinimum(1)
+        spin2.setMaximum(max_val)
+        spin2.setValue(10)
+        spin2.setMinimumWidth(
+            self.fontMetrics().width("X") * (len(str(max_val)) + 1) + 20
+        )
+        spin2.valueChanged.connect(partial(self._spin_changed, spin2))
+
+        self.frequent_rel_spin = spin3 = QDoubleSpinBox()
+        spin3.setMinimum(0)
+        spin3.setDecimals(1)
+        spin3.setSingleStep(0.1)
+        spin3.setMaximum(100)
+        spin3.setValue(10)
+        spin3.setMinimumWidth(self.fontMetrics().width("X") * (2 + 1) + 20)
+        spin3.setSuffix(" %")
+        spin3.valueChanged.connect(partial(self._spin_changed, spin3))
+
+        self.n_values_spin = spin4 = QSpinBox()
+        max_val = min(10, len(variable.categories))
+        spin4.setMinimum(0)
+        spin4.setMaximum(len(variable.categories))
+        spin4.setValue(max_val)
+        spin4.setMinimumWidth(
+            self.fontMetrics().width("X") * (len(str(max_val)) + 1) + 20
+        )
+        spin4.valueChanged.connect(partial(self._spin_changed, spin4))
+
+        grid_layout = QGridLayout()
+        # first row
+        grid_layout.addWidget(radio1, 0, 0, 1, 2)
+        # second row
+        grid_layout.addWidget(radio2, 1, 0, 1, 2)
+        grid_layout.addWidget(spin2, 1, 2)
+        grid_layout.addWidget(label2, 1, 3)
+        # third row
+        grid_layout.addWidget(radio3, 2, 0, 1, 2)
+        grid_layout.addWidget(spin3, 2, 2)
+        grid_layout.addWidget(label3, 2, 3)
+        # fourth row
+        grid_layout.addWidget(radio4, 3, 0)
+        grid_layout.addWidget(spin4, 3, 1)
+        grid_layout.addWidget(label4, 3, 2, 1, 2)
+
+        group_box = QGroupBox("Grouping strategy")
+        group_box.setLayout(grid_layout)
+
+        # grouped variable name
+        new_name_label = QLabel("New value name: ")
+        self.new_name_line_edit = n_line_edit = QLineEdit("Others")
+        # it is shown gray when user removes the text and let user know that
+        # word others is default one
+        n_line_edit.setPlaceholderText("Others")
+        name_hlayout = QHBoxLayout()
+        name_hlayout.addWidget(new_name_label)
+        name_hlayout.addWidget(n_line_edit)
+
+        # confirm_button = QPushButton("Apply")
+        # cancel_button = QPushButton("Cancel")
+        buttons = QDialogButtonBox(
+            orientation=Qt.Horizontal,
+            standardButtons=(QDialogButtonBox.Ok | QDialogButtonBox.Cancel),
+            objectName="dialog-button-box",
+        )
+        buttons.accepted.connect(self.accept)
+        buttons.rejected.connect(self.reject)
+
+        # join components
+        self.setLayout(QVBoxLayout())
+        self.layout().addWidget(group_box)
+        self.layout().addLayout(name_hlayout)
+        self.layout().addWidget(buttons)
+        self.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
+
+    def _spin_changed(self, spin: QSpinBox) -> None:
```

Comment:
Writing three separate functions instead of this one would actually be shorter (excluding docstring, which is not really needed for a one-line function). :)